lib/core/include/dstream.hpp

```diff
@@ -163,6 +230,26 @@ namespace irods::experimental::io
             return transport_->file_descriptor();;
         }
 
+        std::string resource_name() const
+        {
+            return transport_->resource_name();
+        }
+
+        std::string resource_hierarchy() const
+        {
+            return transport_->resource_hierarchy();
+        }
+
+        int replica_number() const
+        {
+            return transport_->replica_number();
+        }
+
+        std::string replica_token() const
+        {
+            return transport_->replica_token();
+        }
+
```

Comment:
Review whether these functions are named properly and needed at all. Use better return types.